pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -1022,6 +1025,28 @@ public ZooKeeper getZkClient() {
         return this.localZooKeeperConnectionProvider.getLocalZooKeeper();
     }
 
+    // get bookkeeper's zookeeper
+    public ZooKeeper getBookieZkClient() {
```

Comment:
I would name this method "createBookieZkClient" because you are actually creating it

Reply:
done.

Reply:
the `bookieZkClient` will be change root to ip:port/ledgers? and the following operation such as `bookieZk().setData(ZkBookieRackAffinityMapping.BOOKIE_INFO_ROOT_PATH,...)` will set data to ip:port/ledgers/bookies path?